### PR TITLE
Added a servletregistration customizer and improved examples.

### PR DESCRIPTION
**Pull request by valdar**

### CODE REVIEW

File: camel-k-runtime-servlet/src/main/java/org/apache/camel/k/servlet/ServletRegistrationContextCustomizer.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.component.servlet.CamelHttpTransportServlet;
+import org.apache.camel.k.ContextCustomizer;
+import org.apache.camel.k.Runtime;
+
+public class ServletRegistrationContextCustomizer implements ContextCustomizer {
+    public static final String DEFAULT_PATH = "/camel/*";
+
+    private String path;
```
Comment (lburgazzoli) [inline camel-k-runtime-servlet/src/main/java/org/apache/camel/k/servlet/ServletRegistrationContextCustomizer.java]: we need getter and setter for this

File: camel-k-runtime-servlet/src/main/java/org/apache/camel/k/servlet/ServletRegistrationContextCustomizer.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    public ServletRegistrationContextCustomizer(String path){
+        this.path = path;
+    }
+
+    @Override
+    public void apply(CamelContext camelContext, Runtime.Registry registry) {
+        registry.bind(
+                "camel-servlet",
+                new ServletRegistration("CamelServlet", new CamelHttpTransportServlet(), path)
```
Comment (lburgazzoli) [inline camel-k-runtime-servlet/src/main/java/org/apache/camel/k/servlet/ServletRegistrationContextCustomizer.java]: we need ot be able to customize the name of the servlet, eventually we can use the name of the key used to bind the ServletRegistration to the camel registry

Comment (valdar) [reply camel-k-runtime-servlet/src/main/java/org/apache/camel/k/servlet/ServletRegistrationContextCustomizer.java]: done

File: camel-k-runtime-servlet/src/main/java/org/apache/camel/k/servlet/ServletRegistrationContextCustomizer.java (hunk 1 above)
Comment (valdar) [reply camel-k-runtime-servlet/src/main/java/org/apache/camel/k/servlet/ServletRegistrationContextCustomizer.java]: done

**Pull request merged**